core/runtime/src/main/java/io/quarkus/runtime/logging/BannerFormatter.java

```diff
@@ -0,0 +1,62 @@
+package io.quarkus.runtime.logging;
+
+import java.util.function.Supplier;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+
+import org.jboss.logmanager.ExtLogRecord;
+import org.jboss.logmanager.formatters.ColorPatternFormatter;
+import org.jboss.logmanager.formatters.FormatStep;
+import org.jboss.logmanager.formatters.PatternFormatter;
+
+public class BannerFormatter extends ColorPatternFormatter {
+
+    private final Supplier<String> bannerSupplier;
+    private Formatter formatter;
+    private boolean isColorPattern;
+
+    BannerFormatter(Formatter formatter, boolean isColorPattern, Supplier<String> bannerSupplier) {
+        this.formatter = formatter;
+        this.isColorPattern = isColorPattern;
+        this.bannerSupplier = bannerSupplier;
+
+        if (isColorPattern) {
+            this.setPattern(((ColorPatternFormatter) formatter).getPattern());
+        } else {
+            this.setPattern(((PatternFormatter) formatter).getPattern());
+        }
+    }
+
+    @Override
+    public String formatMessage(LogRecord logRecord) {
+        if (isColorPattern) {
+            return super.formatMessage(logRecord);
+        } else {
+            return formatter.format(logRecord);
+        }
+    }
+
+    @Override
+    public void setSteps(FormatStep[] steps) {
+        if (isColorPattern) {
+            super.setSteps(steps);
+        } else {
+            ((PatternFormatter) formatter).setSteps(steps);
+        }
+    }
+
+    @Override
+    public String format(ExtLogRecord extLogRecord) {
+        if (isColorPattern) {
+            return ((ColorPatternFormatter) formatter).format(extLogRecord);
+        } else {
+            return ((PatternFormatter) formatter).format(extLogRecord);
+        }
+    }
+
+    @Override
+    public String getHead(Handler h) {
+        return bannerSupplier.get();
```

Comment:
Better check if bannerSupplier is null and call `super.getHead(h)` if it does

Reply:
But creating the instance of bannerFormatter supplier is necessary. So it wouldn't be null in case of bannerFormatter

Reply:
In this case, better make sure bannerSupplier is never null in the constructor

Reply:
I checked the quarkus application without banner extension. It was throwing NPE. It was careless  of me to miss that thing. Thanks for pointing that out. In case banner-extension is not used, application will use normal formatters now